R/github-pat.R

```diff
@@ -44,19 +69,29 @@ set_github_pat <- function(force_new = FALSE, validate = interactive(), verbose
       }
       if(isTRUE(validate)) {
         hx <- curl::handle_setheaders(curl::new_handle(), Authorization = paste("token", cred$password))
-        req <- curl::curl_fetch_memory("https://api.github.com/user", handle = hx)
+        host <- parse_url(url)["host"]
+        api_url <- switch(
+          host,
+          `github.com` = "https://api.github.com/user",
+          sprintf("https://%s/api/v3/user", host)
+        )
+        req <- curl::curl_fetch_memory(api_url, handle = hx)
         if(req$status_code >= 400){
+          # TODO: ideally we would reveal more about the error; we have the info
           message("Authentication failed. Token invalid.")
           credential_reject(cred)
           next
         }
         if(verbose == TRUE){
           data <- jsonlite::fromJSON(rawToChar(req$content))
           helper <- tryCatch(credential_helper_get()[1], error = function(e){"??"})
-          message(sprintf("Using GITHUB_PAT from %s (credential helper: %s)", data$name, helper))
+          message(sprintf(
+            "Using GITHUB_PAT for %s@%s (credential helper: %s)",
+            data$login, cred$host, helper
+          ))
         }
       }
-      return(Sys.setenv(GITHUB_PAT = cred$password))
+      return(Sys.setenv(GITHUB_PAT = cred$password, GITHUB_PAT_USER = cred$username))
```

Comment:
Do you think we should set this? If we're going to consult it on the way in (still not implemented), it feels like we should set it on the way out.